tests/code/test_signinKiosk.py

```diff
@@ -0,0 +1,13 @@
+import pytest
+from app.logic.signinKiosk import sendkioskData
+from app.models.eventParticipant import EventParticipant
+
+@pytest.mark.integration
+def test_sendKioskDataKiosk():

```

Comment:
This doesn't cover all of the cases that the function handles, and doesn't clean up the data after it.

Reply:
You're cleaning up now, which is good, but you're still only exercising one of the two paths. You either update or insert in this function and the test only checks one.